### DISCUSSION TITLE
Weather widget,  age verification issue.

### DISCUSSION THREAD

**Discussion by chip33** [valinet/ExplorerPatcher#838] [Q&A]
I recently have this issue when I open the weather widget

![Screenshot 2022-02-11 190419](https://user-images.githubusercontent.com/48499740/153657251-2192e96f-c32b-42d3-9e92-cc51aaf2d9c5.png)

If I try the highlighted sign in option I get this,

![Screenshot 2022-02-11 191312](https://user-images.githubusercontent.com/48499740/153654903-99897bdc-8a40-4994-a96f-ae28090b1644.png)

 The website for google weather is still working fine in google chrome and ms edge.

**Comment by valinet** [comment]
Yeah, idk, never saw that, didn’t know such a thing existed until 3 seconds ago.

You can try resetting the widget to stock config by deleting the %APPDATA%\ep_weather_host folder. That contains the WebView2 profile for ExplorerPatcher weather. Before that, you can zip it and post it here, so I can more easily try to reproduce and provide a fix. It contains no personal data other than the widget data (city, temperature unit etc) unless you signed into Google account there which should pretty much be impossible since I did not provide an option for that and the widget UI doesn’t show any ‘sign in’ button afaik.

Thanks for the feedback.

**Reply by chip33** [reply]
Deleting the %APPDATA%\ep_weather_host folder fixed it.

There is a sign in button in blue in the image above. it does do something as I clicked it and got the second message. 
I don't feel safe posting anything google account or browser related just in case.

**Reply by chip33** [reply]
If there are single files from %APPDATA%\ep_weather_host that might hold good data I can post them as I can check them first for personal info.

**Reply by valinet** [reply]
.

**Reply by chip33** [reply]
I have sent the zipped up folder from when the issue existed to your email address. 
To me it is a bit different to posting other things as there is too much data to check easily.
I have sent you windbg reports in the past and try to help fix issues when I can.

**Reply by valinet** [reply]
Yeah, sorry about that, there was a problem with my email as well, only now has it arrived, so I did not know about it. Apologies, maybe I was a bit too harsh, or should have phrased it differently, I was speaking more about things in general, it just happened to be on this post. But yeah, rest assured, as far as I know and it’s officially told by the docs, the app data folder of a WebView2 instance contains only data of that instance, so in this case only thing associated with that Google search for weather, since that browser doesn’t do much otherwise.

I will look into it. Apologies if the previous reply sounded too harsh. Truth is, I am thankful with the help you provided me so far and appreciate the feedback from you. It’s just generally, regarding the community around here, I am kind of let down by the lack of the more hands on involvement with this project, plus the complete ignoring of any written documentation by a lot of people.

Thanks and sorry again, I hope you will understand.

**Reply by valinet** [reply]
Yeah, I have checked it out, seems to be a geographical thing, same as the shitty GDPR banner which only shows when one's in the EU. I have used your folder, it just doesn't show the banner for me. Yeah, idk, maybe if I use a VPN from the same country as you're from, I could get it to show...

**Reply by chip33** [reply]
No worries, glad to help. I'm not sure what I did to make it show that strange message.
I am in UK and as far as I know we don't have stringent age laws on our Google accounts, I do sometimes use a European vpn so maybe that triggered it.

**Reply by valinet** [reply]
Yeah, maybe that freaked it out. Anyway, I could, in all seriousness, add a button to clear the widget data and start fresh in the Properties UI, it could help for other problems as well...

**Reply by chip33** [reply]
Okay, seems like a good idea as for some reason it didn't occur to me to find that and just clear it.

**Reply by chip33** [reply]
I did try the "Refresh weather data" button but that didn't seem to fix it.

**Reply by valinet** [reply]
> I did try the "Refresh weather data" button but that didn't seem to fix it.

Yeah, that just updates the widget right away basically (refreshes the web page, thus loading the latest weather data, same as if you hit F5 in the browser, really). Maybe I should call it "Update weather now" or something like that...?

**Reply by chip33** [reply]
Maybe it is Germany as usually I only use Poland or Germany when I'm testing my vpn.

**Reply by chip33** [reply]
Yes, maybe something that makes clear that one just updates/refreshes the weather data and one wipes the cache for troubleshooting or region change issues.
It could always be changed if this issue gets fixed somehow.

**Reply by valinet** [reply]
Implemented an option to clear the widget's data in pre-release 42.10.

**Reply by chip33** [reply]
I always have windows Location services set to Off with my city manually entered into the default location on the same windows settings page.
Maybe google doesn't like it if your hardcoded location in windows or cookies is different to your ip geolocation when behind a vpn so just deleting those cookies on a weather refresh could be enough to fix most issues instead of deleting whole of %APPDATA%\ep_weather_host.
If you don't get many more reports of problems maybe you won't need the "Clear weather widget local data" in the future.

**Reply by valinet** [reply]
It’s easier to delete the entire folder rather than individual cookies. The button is fine there, it could have a number of uses.

**Reply by chip33** [reply]
Cool.

**Reply by doterik** [reply]
@valinet 
> It’s easier to delete the entire folder rather than individual cookies. The button is fine there, it could have a number of uses.

I can confirm that I have had exactly the same issue 3 times by now and was going to suggest something similar because there is no way to do this automatically, as far as I know. (The folder is locked.)

Until now I have unchecked the weather widget, deleted the folder, and then restarted the widget again. To have this done by a click is great.

I just tried it, without having any issue right now, but got an error.
![image](https://user-images.githubusercontent.com/4037602/153759836-fa6deb55-cb3c-48a4-a46e-813dc85bacba.png)

Do I also have to close the weather widget beforehand?

###### I live in the EU with GDPR...

**Reply by valinet** [reply]
No, it should work without disabling it, but ofc even deleting a folder requires a PhD nowadays. Indeed, it needs to disable it temporarily so that the WebView2 processes quit and free the handle to whatever from the folder they use, otherwise it fails. I do that, why it fails for you, I have no clue. Does it work if you disable the widget beforehand? Anyway, you can compile the app, attach with the Visual Studio debugger and see what error it reports. Yeah…

I don’t dislike the GDPR, I just hate its implementation. Every site popping up some stupid window made using the web a catastrophic experience. The reason why these choices are not configured by the user once at the browser level and then be communicated by the navigator to the web site is beyond my understanding. And regarding GDPR itself, yeah, it has its fair share of issues: for e.g., not publicly displaying public exam results of the candidates is something not even the communists dared to do back in the day (yeah, I live in one of **those** countries).

**Reply by doterik** [reply]
Yes, if I disable the weather widget beforehand, then it works (deletes the folder). I was hoping to spare myself this extra click. 😁

Anyway, this is far better than manually having to delete the folder, thanks! 👍🏼

##### /* May I ask a, maybe dumb, question... Is it possible to use the weather widget also on Win10? I love it and could of course try before I ask... 💖 */

**Reply by valinet** [reply]
> Yes, if I disable the weather widget beforehand, then it works (deletes the folder). I was hoping to spare myself this extra click. 😁

It should work without the extra click, as I said. The damn handle is not closed, for whatever reason, even though I disable the widget in the code, which should close the processes that host it, thus freeing it, and I wait for that to happen. I will insert some more dumb sleeps in the hope that the delay is enough for the handle to be closed and for this stupid OS to allow deleting the folder. Basically wait more.

> ##### /* May I ask a, maybe dumb, question... Is it possible to use the weather widget also on Win10? I love it and could of course try before I ask... 💖 */

Some colleague at work which sticks to Windows 10 LTSC releases asked me the same thing on Friday. Yeah, I intend to make it somehow work there as well. Easiest would be to have ExplorerPatcher work in Windows 10 as well, basically guarding all Windows 11-related stuff. Yeah, idk, I will look into it, I will think of something.

**Reply by valinet** [reply]
As for trying it right now on Windows 10: I recommend you don't do that. As it is right now, EP doesn't work on Windows 10 and will brick `explorer`, so if you do that, be prepared with the knowledge on how to restore it. Or just don't do it. There are neither checks to prevent you from running EP on Windows 10, nor any guards in the code to not access the new things that are not there in Windows 10. It just crashes when it doesn't find them, afaik.

**Reply by doterik** [reply]
Yes, I suppose you're right, the handles do not have enough time to close, at least not on my machine... I can live with that...

Thanks for the answer, and warning, about Win10... I will **_not_** try then...

**Reply by doterik** [reply]
With the latest version (.11), `clear local data` now works like a charm, thank you. 👏🏼

**Reply by valinet** [reply]
> With the latest version (.11), `clear local data` now works like a charm, thank you. 👏🏼

That's great to hear, thanks for the feedback.
